### PR TITLE
Adds the media plugin to TinyMCE to allow insertion of videos.

### LINKED ISSUE #266: Feature Request: Add ability to insert video
### For Feature Requests
It would be great to add the ability to insert video in pages. I did this using tinymce media plugin, but little uncomfortably.

### PR DESCRIPTION
**Pull request by Abijeet**
Fixes #266

Signed-off-by: Abijeet <abijeetpatro@gmail.com>

screenshot-2018-3-28 editing page new page bookstack

I've had to add an ugly hacky fix to remove the "Advanced" tab and the "Open" button,
https://github.com/BookStackApp/BookStack/commit/6f280bb4884b2d693d89e84e69b94f274b9747f3#diff-4fd06e033e83fb244f4d8c9752a26781R315

Otherwise this is how the default media dialog looks like with the open button not really doing anything.

screenshot-2018-3-28 editing page new page bookstack 1

### CODE REVIEW

File: resources/assets/sass/_pages.scss [hunk 1]
```diff
@@ -309,4 +309,14 @@
...
   min-height: 175px;
+}
+
+/* FIXME - Ugly hack to modify the media plugin for TinyMCE */
+.mce-floatpanel[aria-label="Insert/edit media"] {
+	.mce-open {
+		display: none;
+	}
+	.mce-tab:last-of-type {
+		display: none;
```
Comment (ssddanbrown) [inline resources/assets/sass/_pages.scss]: From a quick test not sure this rule to hide the tab is needed, I think the tab hides itself when both:

```
media_alt_source: false,
media_poster: false,
```
Are set.

Only quickly tested in codepen myself though.

**Comment by Abijeet** [comment]
@ssddanbrown  - Thanks for the quick review. I did try that, but the advanced tab was still visible. As per the [changelog](https://www.tinymce.com/docs/changelog/#version477february192018) this was fixed in TinyMCE Version 4.7.7 released on February 19, 2018.

Search for - 
> fixed a bug where the advanced tab in the Media Plugin was being shown even if it didn't contain anything. Patch contributed by gabrieeel.

We are using version 4.6.6 (2017-08-30).

Maybe its time to upgrade?

**Comment by ssddanbrown** [comment]
@Abijeet ah, Makes sense. Yeah, Upgrading is probably a good idea, Think it's been a few releases since last TinyMCE upgrade.

**Comment by ssddanbrown** [comment]
@Abijeet Sorry but I've just made this a little more difficult to merge after pushing 736d7118b08ef7f03d35a237bf5a909ada90d9cd up. 

Will need to move the `resources/assets/js/pages/page-form.js` changes to `resources/assets/js/components/wysiwyg-editor.js`.

**Comment by Abijeet** [comment]
@ssddanbrown - No problem, will take care of it this weekend.

**Comment by mendiromania** [comment]
rip

**Comment by Abijeet** [comment]
@ssddanbrown - Done!

**Comment by ssddanbrown** [comment]
@Abijeet Awesome, Thanks!

**Pull request merged**